### PR TITLE
gnrc/pktdump : Expose Configurations to Kconfig

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in PKTDUMP GNRC to Kconfig.

### Testing procedure

1. New documentation was built using doxygen 

The build works fine.

2. Test folder and files were introduced to show the macro on Native environment.

[Test Files](https://github.com/akshaim/RIOT/commit/e1e224a59e5dcd6b329cf10161ff9e191d273680) - **Updated**

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-608-g1171d8-Kconfig_pktdump_tests)
CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP=3
GNRC_PKTDUMP_MSG_QUEUE_SIZE=(1<<3)

#### Usage with CFLAGS 

[/tests/gnrc_pkdump/Makefile](https://github.com/akshaim/RIOT/commit/e1e224a59e5dcd6b329cf10161ff9e191d273680)

> CFLAGS += -DCONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP=4

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-608-g1171d8-Kconfig_pktdump_tests)
CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP=4
GNRC_PKTDUMP_MSG_QUEUE_SIZE=(1<<4)

#### Usage with Kconfig

[/tests/gnrc_pktdump](https://github.com/akshaim/RIOT/commit/e1e224a59e5dcd6b329cf10161ff9e191d273680)

> make menuconfig

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-608-g1171d8-Kconfig_pktdump_tests)
CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP=7
GNRC_PKTDUMP_MSG_QUEUE_SIZE=(1<<7)

### Issues/PRs references

#12888

### CODE REVIEW

**Comment by miri64** [comment]
Mh.. as far as I can see, except for `gnrc_netif` the queue sizes in GNRC were not touched so far by Kconfig. @leandrolanzieri is there a reason for that?

**Comment by leandrolanzieri** [comment]
> Mh.. as far as I can see, except for `gnrc_netif` the queue sizes in GNRC were not touched so far by Kconfig. @leandrolanzieri is there a reason for that?

Which queue sizes are missing from the exposed modules? I see these configurable queue sizes:
- `GNRC_SIXLOWPAN_MSG_QUEUE_SIZE`
- `GNRC_IPV6_MSG_QUEUE_SIZE`
- `GNRC_NETIF_MSG_QUEUE_SIZE`

**Comment by miri64** [comment]
> Which queue sizes are missing from the exposed modules? I see these configurable queue sizes:
> 
>     * `GNRC_SIXLOWPAN_MSG_QUEUE_SIZE`
> 
>     * `GNRC_IPV6_MSG_QUEUE_SIZE`
> 
>     * `GNRC_NETIF_MSG_QUEUE_SIZE`

TCP, UDP and RPL should also be configurable. I must have looked at an older branch, because now I can confirm, that IPv6 and 6LoWPAN are configurable via Kconfig.

**Comment by leandrolanzieri** [comment]
> TCP, UDP and RPL should also be configurable. I must have looked at an older branch, because now I can confirm, that IPv6 and 6LoWPAN are configurable via Kconfig.

RPL is waiting for review #13941, the other modules are yet pending to be exposed.

File: sys/include/net/gnrc/pktdump.h [hunk 1]
```diff
@@ -28,13 +28,23 @@
 extern "C" {
 #endif
 
+/**
+ * @defgroup net_gnrc_pktdump_conf GNRC PKTDUMP compile configurations
+ * @ingroup net_gnrc_conf
+ * @{
+ */
 /**
  * @brief   Message queue size for the pktdump thread
```
Comment (miri64) [inline sys/include/net/gnrc/pktdump.h:37]: This should be moved done to the queue size definition, not the exponent.

File: sys/net/gnrc/pktdump/Kconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    bool "Configure GNRC_PKTDUMP"
+    depends on MODULE_GNRC_PKTDUMP
+    help
+        Configure the GNRC_PKTDUMP using Kconfig.
+
+if KCONFIG_MODULE_GNRC_PKTDUMP
+
+config GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP
+    int "Queue size (as exponent of 2^n)"
+    default 4
```
Comment (miri64) [inline sys/net/gnrc/pktdump/Kconfig:17]: Shouldn't this be 3, given that `GNRC_PKTDUMP_QUEUE_SIZE` is currently 8?

File: sys/include/net/gnrc/pktdump.h [hunk 3]
```diff
@@ -28,13 +28,23 @@
...
  */
-#ifndef GNRC_PKTDUMP_MSG_QUEUE_SIZE
-#define GNRC_PKTDUMP_MSG_QUEUE_SIZE     (8U)
+#ifdef CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP
+#define CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE  (1<<CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP)
```
Comment (miri64) [inline sys/include/net/gnrc/pktdump.h:40]: Style
```suggestion
#define CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE \
    (1 << CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP)
```

File: sys/net/gnrc/pktdump/Kconfig [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+menuconfig KCONFIG_MODULE_GNRC_PKTDUMP
+    bool "Configure GNRC_PKTDUMP"
+    depends on MODULE_GNRC_PKTDUMP
+    help
+        Configure the GNRC_PKTDUMP using Kconfig.
+
+if KCONFIG_MODULE_GNRC_PKTDUMP
+
+config GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP
+    int "Queue size (as exponent of 2^n)"
```
Comment (miri64) [inline sys/net/gnrc/pktdump/Kconfig:16]: I find this text misleading. It is not the queue size, but the exponent *for* the queue size:

```suggestion
    int "Exponent for the queue size (resulting in the queue size 2^n)"
```

File: sys/net/gnrc/pktdump/Kconfig [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    default 4
+    help
+        Message queue size for the pktdump thread. Value represents the
```
Comment (miri64) [inline sys/net/gnrc/pktdump/Kconfig:19]: Exponent for the message queue size...

File: sys/include/net/gnrc/pktdump.h (hunk 3 above)
Comment (miri64) [inline sys/include/net/gnrc/pktdump.h:40]: Does this need to be renamed to `CONFIG_` if it is not exposed to Kconfig?

Comment (leandrolanzieri) [reply sys/include/net/gnrc/pktdump.h:40]: Well, if this macro is not configurable then the prefix is not needed. But also, the `ifndef` can be removed as the user would configure the queue size via `CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP`

File: sys/net/gnrc/pktdump/Kconfig (hunk 2 above)
Comment (akshaim) [reply sys/net/gnrc/pktdump/Kconfig:17]: Yes. I shall correct it

File: sys/include/net/gnrc/pktdump.h (hunk 3 above)
Comment (akshaim) [reply sys/include/net/gnrc/pktdump.h:40]: Fine. I shall update the header based on PR #14086

Comment (akshaim) [reply sys/include/net/gnrc/pktdump.h:40]: This strangely results in error. Hence I did now wrap. Also I think its in 100 char limit.

File: sys/net/gnrc/pktdump/Kconfig (hunk 5 above)
Comment (akshaim) [reply sys/net/gnrc/pktdump/Kconfig:19]: Shall update.

**Comment by miri64** [comment]
Needs rebase

**Comment by akshaim** [comment]
> Needs rebase

Done. Updated and tested.

File: sys/net/gnrc/pktdump/Kconfig [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_MODULE_GNRC_PKTDUMP
+    bool "Configure GNRC_PKTDUMP"
```
Comment (leandrolanzieri) [inline sys/net/gnrc/pktdump/Kconfig:8]: Here and bellow:
```suggestion
    bool "Configure GNRC Packet Dump"
```

**Comment by leandrolanzieri** [comment]
@miri64 are you OK with the changes?

**Comment by miri64** [comment]
@akshaim please squash!

**Comment by akshaim** [comment]
@miri64 I have retained the #ifndef guards as discussed in PR #14086

File: sys/include/net/gnrc/pktdump.h [hunk 7]
```diff
@@ -28,11 +28,29 @@
...
+#define CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP  3
+#endif
+/** @} */
+
 /**
  * @brief   Message queue size for the pktdump thread
  */
 #ifndef GNRC_PKTDUMP_MSG_QUEUE_SIZE
-#define GNRC_PKTDUMP_MSG_QUEUE_SIZE     (8U)
+#define GNRC_PKTDUMP_MSG_QUEUE_SIZE  (1<<CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP)
```
Comment (miri64) [inline sys/include/net/gnrc/pktdump.h:53]: Minor style-nit somehow not caught by our code beautifiers:

```suggestion
#define GNRC_PKTDUMP_MSG_QUEUE_SIZE  (1 << CONFIG_GNRC_PKTDUMP_MSG_QUEUE_SIZE_EXP)
```

From the Linux style guide we inherit from ([section 3.1]()):

```rst
Use one space around (on each side of) most binary and ternary operators,
such as any of these::

	=  +  -  <  >  *  /  %  |  &  ^  <=  >=  ==  !=  ?  :
```

**Comment by miri64** [comment]
You may squash the style fix immediately

File: sys/include/net/gnrc/pktdump.h (hunk 7 above)
Comment (akshaim) [reply sys/include/net/gnrc/pktdump.h]: Sure. I was trying to keep the line under 80 chars hence skipped the spaces.

**Comment by akshaim** [comment]
> You may squash the style fix immediately

Done.

File: sys/include/net/gnrc/pktdump.h (hunk 7 above)
Comment (miri64) [reply sys/include/net/gnrc/pktdump.h]: 80 is the soft limit. Line length should not go over 100 though :-).

Comment (akshaim) [reply sys/include/net/gnrc/pktdump.h]: Okay

**Pull request merged**